goenvtemplator.go

```diff
@@ -52,46 +67,115 @@ func (ef *envFiles) String() string {
 	return fmt.Sprintf("%v", *ef)
 }
 
-func generateTemplates(ts templatesPaths, debug bool, delimLeft string, delimRight string) error {
+func writeFile(destinationPath string, data string) error {
+	if !filepath.IsAbs(destinationPath) {
+		log.Fatalf("Destination path '%s' is not absolute!", destinationPath)
+		return errors.New("absolute path error")
+	}
+
+	if err := ioutil.WriteFile(destinationPath, []byte(data), 0664); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func readSource(templatePath string) (string, error) {
+	if !filepath.IsAbs(templatePath) {
+		log.Fatalf("Template path '%s' is not absolute!", templatePath)
+		return "", errors.New("absolute path error")
+	}
+
+	var slice []byte
+	var err error
+	if slice, err = ioutil.ReadFile(templatePath); err != nil {
+		return "", err
+	}
+
+	return string(slice), nil
+
+}
+
+func Debug(message string, args ...interface{}) {
+	if DEBUG {
+		log.Printf(message, args...)
+	}
+}
+
+func generateTemplates(
+	ts templatesPaths,
+	delimLeft string,
+	delimRight string,
+	engine string) error {
+
 	for _, t := range ts {
 		if v > 0 {
 			log.Printf("generating %s -> %s", t.source, t.destination)
 		}
-		if err := generateFile(t.source, t.destination, debug, delimLeft, delimRight); err != nil {
-			return fmt.Errorf("Error while generating '%s' -> '%s'. %v", t.source, t.destination, err)
+
+		var templar Templar
+
+		source, err := readSource(t.source)
+		if err != nil {
+			return err
+		}
+
+		templateName := filepath.Base(t.source)
+
+		switch engine {
+		case "pongo":
+			templar = &PongoTemplar{
+				Source: source,
+			}
+		default:
```

Comment:
What about failing in default and have a `case "text/template"`?

Reply:
That breaks backwards compatibility. You would have to change every Dockerfile with goenvtemplator entrypoint... for example.

Reply:
No, I mean 
```
+	flag.StringVar(&engine, "engine", "text/template", "Override default text/template [support: pongo2]")
```

and to have a `case "text/template"` and default if an user passes `--engine=foo-bar` to fail and not to use default `text/template` engine.

Reply:
Truly looks better and more deterministic. Will fix.